File: node/examples/basic-bot.ts
import { WebexMessageHandler, consoleLogger } from '../src/index.js';

const handler = new WebexMessageHandler({
  token: process.env.WEBEX_BOT_TOKEN!,
  logger: consoleLogger,
});

handler.on('message:created', (msg) => {
  console.log(`[${msg.personEmail}] ${msg.text}`);
  if (msg.html) {
    console.log(`  HTML: ${msg.html}`);
  }
});

handler.on('message:deleted', (data) => {
  console.log(`Message ${data.messageId} deleted by ${data.personId}`);
});

handler.on('connected', () => console.log('Connected to Webex'));
handler.on('disconnected', (reason) => console.log(`Disconnected: ${reason}`));
handler.on('reconnecting', (attempt) => console.log(`Reconnecting (attempt ${attempt})...`));
handler.on('error', (err) => console.error('Error:', err.message));

// Graceful shutdown
process.on('SIGINT', async () => {
  console.log('Shutting down...');
  await handler.disconnect();
  process.exit(0);
});

handler.connect().catch((err) => {
  console.error('Failed to connect:', err);
  process.exit(1);
});

File: python/examples/basic_bot.py
"""Basic Webex bot example using webex-message-handler."""

import asyncio
import logging
import os
import signal

from webex_message_handler import WebexMessageHandler, WebexMessageHandlerConfig, console_logger

logger = logging.getLogger("basic_bot")
logging.basicConfig(level=logging.INFO, format="%(levelname)s %(name)s: %(message)s")


async def main() -> None:
    token = os.environ.get("WEBEX_BOT_TOKEN")
    if not token:
        print("Set WEBEX_BOT_TOKEN environment variable")
        return

    handler = WebexMessageHandler(
        WebexMessageHandlerConfig(token=token, logger=console_logger)
    )

    @handler.on("message:created")
    async def on_message(msg):
        print(f"[{msg.person_email}] {msg.text}", flush=True)
        if msg.html:
            print(f"  HTML: {msg.html}", flush=True)

    @handler.on("message:deleted")
    def on_deleted(data):
        print(f"Message {data.message_id} deleted by {data.person_id}")

    @handler.on("connected")
    def on_connected():
        print("Connected to Webex", flush=True)

    @handler.on("disconnected")
    def on_disconnected(reason):
        print(f"Disconnected: {reason}", flush=True)

    @handler.on("reconnecting")
    def on_reconnecting(attempt):
        print(f"Reconnecting (attempt {attempt})...", flush=True)

    @handler.on("error")
    def on_error(err):
        print(f"Error: {err}", flush=True)

    # Graceful shutdown
    loop = asyncio.get_event_loop()
    stop_event = asyncio.Event()

    def _shutdown():
        stop_event.set()

    for sig in (signal.SIGINT, signal.SIGTERM):
        try:
            loop.add_signal_handler(sig, _shutdown)
        except NotImplementedError:
            # Windows doesn't support add_signal_handler
            signal.signal(sig, lambda s, f: _shutdown())

    try:
        await handler.connect()
        await stop_event.wait()
    except KeyboardInterrupt:
        pass
    finally:
        print("Shutting down...")
        await handler.disconnect()


if __name__ == "__main__":
    asyncio.run(main())
